From 6b642d79378081ddd3aecc0b19477f6ee00a7916 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 16 Jun 2020 21:31:03 +0200
Subject: [PATCH 57/N] Mirror master and main

In preparation for eventually switching over to `main`, let's
synchronize the historical and the designated future main branch.

This follows the excellent advice from
https://github.com/chancancode/branch-rename/#gradual-migration

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 .github/workflows/mirror-master-and-main.yml | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)
 create mode 100644 .github/workflows/mirror-master-and-main.yml

diff --git a/.github/workflows/mirror-master-and-main.yml b/.github/workflows/mirror-master-and-main.yml
new file mode 100644
index 0000000..8516479
--- /dev/null
+++ b/.github/workflows/mirror-master-and-main.yml
@@ -0,0 +1,19 @@
+name: Mirror "master" and "main" branches
+on:
+  push:
+    branches:
+      - master
+      - main
+
+jobs:
+  mirror:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Partial clone
+        env:
+          ref: ${{ github.event.ref }}
+        run: git clone --bare --depth=100 --single-branch --branch ${ref#refs/heads/} --filter=blob:none ${{ github.event.repository.html_url }} .
+      - name: Push
+        run: |
+          git config http.https://github.com/.extraheader "Authorization: Basic $(echo -n x-access-token:${{ github.token }} | base64 --wrap=0)"
+          git push origin HEAD:master HEAD:main
-- 
2.9.0

